updatePromotionProcess.php

```diff
@@ -0,0 +1,63 @@
+<?php
+require 'connection.php';
+
+if (isset($_POST["jsonText"])) {
+    $jsonText = $_POST["jsonText"];
+    $dataObject = json_decode($jsonText);
+
+    $proID = $dataObject->proID;
+    $name = $dataObject->name;
+    $disciption = $dataObject->discription;
+    $discount = $dataObject->discount;
+    $startDate = $dataObject->startDate;
+    $endDate = $dataObject->endDate;
+    $imageFile = null;
+    $status = "no";
+
+    date_default_timezone_set('Asia/Colombo');
+    $currentDate = date('Y-m-d H:i:s');
+
+    if (isset($_FILES["imagePath"])) {
```

Comment:
Put that code fragment after the `UPDATE` process. In this case, there is some possibility that an `error` will occur during the update process. So the new data will **not be saved** to the database, but the image will **already be saved** to the file path. That's not appropriate...